src/devices_models/devices/common/device_range_constraints.jl

```diff
@@ -107,23 +113,32 @@ function device_range_constraints!(
     end
 
     for rc in range_constraints
-        constraint_infos = Vector{DeviceRangeConstraintInfo}(undef, length(devices))
+        constraint_struct = rc.constraint_struct
+        constraint_infos = Vector{constraint_struct}(undef, length(devices))
         cons_name = constraint_name(rc.constraint_name, T)
         var_name = variable_name(rc.variable_name, T)
         if var_name in ff_affected_variables
             @debug "Skip adding $var_name because it is handled by feedforward"
             continue
         end
-        bin_var_name = isnothing(rc.bin_variable_name) ? rc.bin_variable_name :
-            variable_name(rc.bin_variable_name, T)
+        bin_var_name = isempty(rc.bin_variable_names) ? rc.bin_variable_names :
+            [variable_name(name, T) for name in rc.bin_variable_names]
         for (i, dev) in enumerate(devices)
             dev_name = PSY.get_name(dev)
             limits = rc.limits_func(dev)
             if isnothing(limits)
                 limits = (min = 0.0, max = 0.0)
                 @warn "Range constraint limits of $T $dev_name are nothing. Set to" limits
             end
-            constraint_info = DeviceRangeConstraintInfo(dev_name, limits)
+            if constraint_struct == DeviceRangeConstraintInfo
+                constraint_info = DeviceRangeConstraintInfo(dev_name, limits)
+            elseif constraint_struct == DeviceMultiStartRangeConstraintsInfo
+                lag_limits = rc.lag_limits_func(dev)
+                constraint_info =
+                    DeviceMultiStartRangeConstraintsInfo(dev_name, limits, lag_limits)
+            else
+                error("Missing implementation for $constraint_struct")
+            end
```

Comment:
looks like this block should be a dispatched function not an if-else conditional.

Reply:
That would be better.  But the different constraint structs take different lists of parameters.